app.rb

```diff
@@ -1,21 +1,44 @@
 require './config/environment'
-require './models/users'
+require './models/user'
 require 'sinatra/base'
 require 'sinatra'
 require 'sinatra/reloader' if development?
 
 class Mkbnb < Sinatra::Base
+  enable :sessions
   configure :development do
     register Sinatra::Reloader
   end
 
+  before do
+    @signed_in_user = session[:current_user]
+  end
+
   get '/' do
-    @users = User.all
-    
-    # @users.each { |user| p user.first_name } 
     erb :index
   end
 
+  get '/login' do
+    erb :login
+  end
+
+  post '/login' do
+    session[:current_user] = User.can_log_in(params[:user_name], params[:password])
+    if session[:current_user] == nil
+      redirect '/invalid_login'
+    else
+      redirect '/explore_rooms'
+    end
```

Comment:
maybe more succinct/readable to write this as 

`session[:current_user].nil? ? redirect '/invalid_login' : redirect '/explore_rooms'`